### PR TITLE
Determine a suitable initial size for the Stage #367

### LINKED ISSUE #367: Determine a suitable initial size for the Stage
As per title.

Continuation from #366, #333

### PR DESCRIPTION
**Pull request by yl-coder**
Fixes #367

### CODE REVIEW

File: src/main/java/address/ui/Ui.java [hunk 1]
```diff
@@ -4,18 +4,23 @@
 import address.browser.BrowserManager;
 import address.controller.MainController;
 import address.model.ModelManager;
+import address.model.UserPrefs;
 import address.util.Config;
 import javafx.scene.control.Alert;
 import javafx.stage.Stage;
+import javafx.util.Pair;
+import org.apache.commons.lang3.tuple.ImmutablePair;
```
Comment (damithc) [inline src/main/java/address/ui/Ui.java]: Any alternative for this? Are we using Apache.commons in other places?

Comment (yl-coder) [inline src/main/java/address/ui/Ui.java]: No only this place. Perhaps I could write my own custom ScreenSize class and implement Serializable.

Comment (damithc) [inline src/main/java/address/ui/Ui.java]: Yes, I'd rather minimize additional libraries unless we really need them.
BTW, don't you need to save the position as well?

File: src/main/java/address/util/ScreenSize.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package address.util;
+
+import java.io.Serializable;
+
+/**
+ * A Serializable class that contains the screen width and height.
+ */
+public class ScreenSize implements Serializable {
```
Comment (damithc) [inline src/main/java/address/util/ScreenSize.java]: Is this screen size or window size?
In future, you might be adding more attributes to this class. e.g. x-position, y-position, width of panel, etc. May be want to give a more general name. e.g. GuiSettings

File: src/main/java/address/util/ScreenUtil.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+/**
+ * A utility class to contain methods related to the Screen.
+ */
+public final class ScreenUtil {
+
+    public static ScreenSize getRecommendedScreenSize() {
+        Rectangle2D primScreenBounds = Screen.getPrimary().getVisualBounds();
+        return new ScreenSize(primScreenBounds.getWidth() * 0.5, primScreenBounds.getHeight() * 0.7);
+    }
```
Comment (damithc) [inline src/main/java/address/util/ScreenUtil.java]: You can put this in the ScreenSize class?

File: src/main/java/address/model/UserPrefs.java [hunk 4]
```diff
@@ -35,4 +40,12 @@ public String getSaveLocationString() {
     public boolean isSaveLocationSet() {
         return getSaveLocation() != null;
     }
+
+    public ScreenSize getScreenSize() {
+        return screenSize == null ? ScreenUtil.getRecommendedScreenSize() : screenSize;
```
Comment (damithc) [inline src/main/java/address/model/UserPrefs.java]: `return screenSzie == null ? new ScreenSize() : screenSize;`
The default constructor will use default values.

Comment (yl-coder) [inline src/main/java/address/model/UserPrefs.java]: noted. changed

File: src/main/java/address/util/ScreenUtil.java (hunk 3 above)
Comment (yl-coder) [inline src/main/java/address/util/ScreenUtil.java]: Yes removed.

File: src/main/java/address/util/ScreenSize.java (hunk 2 above)
Comment (yl-coder) [inline src/main/java/address/util/ScreenSize.java]: Window size, implemented the x and y position.

**Pull request merged**